### ISSUE TITLE
Keras & Atom incompatibility?

### ISSUE THREAD

**Issue by bmnds** [tvdboom/ATOM#8]
Hi, I'm trying to add a Keras model to my atom instance so I can compare it with my other models, but it always recognizes the score as 0.5, very different from what Keras is printing out:
![image](https://user-images.githubusercontent.com/433506/123020658-a3b97300-d3a0-11eb-9fc0-bb90c709e0d0.png)

Any ideas if this is something I'm not configuring properly or a bug?
![image](https://user-images.githubusercontent.com/433506/123020730-c6e42280-d3a0-11eb-8662-9f386ea70215.png)

Here is the snippet of how I build the Keras model:
```Python
input_shape = atom.X_train.shape[1]

def neural_network():
    keras_model = Sequential()
    keras_model.add(Dense(70,input_shape=(input_shape,),activation='relu'))
    keras_model.add(Dropout(0.3))
    keras_model.add(Dense(35, activation='relu'))
    keras_model.add(Dropout(0.3))
    keras_model.add(Dense(2, activation='sigmoid'))
    
    keras_model.compile(loss='categorical_crossentropy',
                  optimizer='Adam',
                  metrics=[AUC(), Recall()])
    
    return keras_model

from keras.wrappers.scikit_learn import KerasClassifier
model = KerasClassifier(neural_network, 
                        epochs=50, 
                        batch_size=128, 
                        verbose=1)

atom.run(model, # Keras
         metric="roc_auc", # ROC AUC
         needs_proba=True,
         random_state=random_state)
```

**Comment by tvdboom** [comment]
Hi. This is weird indeed. The score of 0.5 is correct. The problem is that the estimator is predicting only zeros. Check it with `atom.kc.predict_test`. The question is, why is it predicting only zeros? The problem appears to fix itself when you scale the features with `atom.scale()` before fitting (which I would always recommend when using a MLP), but still, this should not happen. I'll look more into it later.

**Comment by bmnds** [comment]
Damn, you are fast! Thanks again for such a quick response :)
I'll test this later today. I'm still learning how to work with MLP models, so I wasn't aware of the scale thing yet.

**Comment by tvdboom** [comment]
Have you managed to solve the problem? I looked into it but can't find the bug on ATOM's side. It's weird that Keras' scores on your example are different. When I run tests myself they are consistent. What happens in my case is that the model is not learning anything (the loss is not going down, probably because of insufficient data). That's the reason it predicts only zeros and the auc is 0.5. Can you check if that's the case for you? And play around with atom.kc.estimator.predict and compare its results with atom.kc.predict. If those are the same then the problem is with the training of the neural network.

**Comment by bmnds** [comment]
Yes, after applying your suggestion it worked like a charm!

I've  applied the standard scaler:
![image](https://user-images.githubusercontent.com/433506/123330990-b18c0700-d50c-11eb-943a-4a87d809efe9.png)

Oh, I did something else too, I've switched KerasClassifer from `keras.wrappers.scikit_learn` to use the one from `scikeras.wrappers`:
![image](https://user-images.githubusercontent.com/433506/123331206-fd3eb080-d50c-11eb-8a0d-9ea45fa04d0a.png)

Here is the output after those changes:
![image](https://user-images.githubusercontent.com/433506/123331277-121b4400-d50d-11eb-9373-d2dc30321bd2.png)

Now I need to dig deeper into these algorithms and try to find how to optimize these results, thanks a lot for your help :)!

**Comment by tvdboom** [comment]
Glad to hear it worked. To optimize the results you could try hyperparameter tuning. Check the [user guide](https://tvdboom.github.io/ATOM/user_guide/training/#hyperparameter-tuning) and the 
[deep learning example](https://tvdboom.github.io/ATOM/examples/deep_learning/) to apply Bayesian Optimization using custom hyperparameters.
